src/NosCore.Controllers/CharacterScreenPacketController.cs

```diff
@@ -315,6 +318,14 @@ public void SelectCharacter(SelectPacket selectPacket)
                     Session.Character.Mp = (int) Session.Character.MPLoad();
                 }
 
+                IEnumerable<CharacterRelation> relations = DAOFactory.CharacterRelationDAO.Where(s => s.CharacterId == Session.Character.CharacterId).Cast<CharacterRelation>();
+
+                foreach (CharacterRelation relation in relations)
+                {
+                    relation.CharacterName = DAOFactory.CharacterDAO.Where(s => s.CharacterId == relation.RelatedCharacterId)?.FirstOrDefault()?.Name;
```

Comment:
still need to be outside of the foreach, just use a where(in list) and iterate on this saved list